src/internal/connector/support/m365Support_test.go

```diff
@@ -159,3 +160,56 @@ func (suite *DataSupportSuite) TestCreateListFromBytes() {
 		})
 	}
 }
+
+func (suite *DataSupportSuite) TestHasAttachments() {
+	tests := []struct {
+		name          string
+		hasAttachment assert.BoolAssertionFunc
+		getBodyable   func(t *testing.T) models.ItemBodyable
+	}{
+		{
+			name:          "Mock w/out attachment",
+			hasAttachment: assert.False,
+			getBodyable: func(t *testing.T) models.ItemBodyable {
+				byteArray := mockconnector.GetMockMessageWithBodyBytes(
+					"Test",
+					"This is testing",
+					"This is testing",
+				)
+				message, err := CreateMessageFromBytes(byteArray)
+				require.NoError(t, err)
+				return message.GetBody()
+			},
+		},
+		{
+			name:          "Mock w/ inline attachment",
+			hasAttachment: assert.True,
+			getBodyable: func(t *testing.T) models.ItemBodyable {
+				byteArray := mockconnector.GetMessageWithOneDriveAttachment("Test legacy")
+				message, err := CreateMessageFromBytes(byteArray)
```

Comment:
@dadams39 - should add a test to `func (suite *GraphConnectorIntegrationSuite) TestRestoreAndBackup()` also please to cover this e2e. The current test only validates the behavior of `HasAttachments`